pyproject.toml

```diff
@@ -6,17 +6,17 @@ authors = []
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = "^3.8.1"
+python = ">=3.8.1,<3.12"
 langchain = "^0.0.137"
 openai = "^0.27.4"
 pydantic = "^1.10.7"
 faiss-cpu = "^1.7.3"
 tiktoken = "^0.3.3"
 google-search-results = "^2.4.2"
 
-
-[tool.poetry.group.dev.dependencies]
+[tool.poetry.dev-dependencies]
```

Comment:
It seems that you have an outdated poetry version installed (https://python-poetry.org/docs/master/managing-dependencies/).